service/backend/pchain/block.go

```diff
@@ -0,0 +1,431 @@
+		hash = *request.BlockIdentifier.Hash
+	}
+
+	isGenesisBlockRequest, err := b.isGenesisBlockRequest(ctx, blockIndex, hash)
+	if err != nil {
+		return nil, service.WrapError(service.ErrClientError, err)
+	}
+
+	if isGenesisBlockRequest {
+		genesisBlock, transactions, err := b.getGenesisBlockAndTransactions(ctx, request.NetworkIdentifier)
+		if err != nil {
+			return nil, service.WrapError(service.ErrClientError, err)
+		}
+
+		return &types.BlockResponse{
+			Block: &types.Block{
+				BlockIdentifier: b.genesisBlockIdentifier,
+				// Parent block identifier of genesis block is set to itself instead of the hash of the genesis state
+				// This is done as the genesis state hash cannot be used as a transaction id for the /block apis
+				// and the operations found in the genesis state are returned as operations of the genesis block.
+				ParentBlockIdentifier: b.genesisBlockIdentifier,
+				Transactions:          transactions,
+				Timestamp:             genesisBlock.Timestamp,
+				Metadata: map[string]interface{}{
+					pmapper.MetadataMessage: genesisBlock.Message,
+				},
+			},
+		}, nil
+	}
+
+	block, err := b.getBlockDetails(ctx, blockIndex, hash)
+	if err != nil {
+		return nil, service.WrapError(service.ErrClientError, err)
+	}
+	blockIndex = int64(block.Height)
+
+	transactions, err := b.parseTransactions(ctx, request.NetworkIdentifier, block.Txs)
+	if err != nil {
+		return nil, service.WrapError(service.ErrInternalError, err)
+	}
+
+	resp := &types.BlockResponse{
+		Block: &types.Block{
+			BlockIdentifier: &types.BlockIdentifier{
+				Index: blockIndex,
+				Hash:  block.BlockID.String(),
+			},
+			ParentBlockIdentifier: &types.BlockIdentifier{
+				Index: blockIndex - 1,
+				Hash:  block.ParentID.String(),
+			},
+			Timestamp:    block.Timestamp,
+			Transactions: transactions,
+		},
+	}
+
+	return resp, nil
+}
+
+// BlockTransaction implements the /block/transaction endpoint.
+func (b *Backend) BlockTransaction(ctx context.Context, request *types.BlockTransactionRequest) (*types.BlockTransactionResponse, *types.Error) {
+	isGenesisBlockRequest, err := b.isGenesisBlockRequest(ctx, request.BlockIdentifier.Index, request.BlockIdentifier.Hash)
+	if err != nil {
+		return nil, service.WrapError(service.ErrClientError, err)
+	}
+
+	var transactions []*types.Transaction
+
+	if isGenesisBlockRequest {
+		_, transactions, err = b.getGenesisBlockAndTransactions(ctx, request.NetworkIdentifier)
+		if err != nil {
+			return nil, service.WrapError(service.ErrInternalError, err)
+		}
+	} else {
+		block, err := b.getBlockDetails(ctx, request.BlockIdentifier.Index, request.BlockIdentifier.Hash)
+		if err != nil {
+			return nil, service.WrapError(service.ErrClientError, err)
+		}
+
+		transactions, err = b.parseTransactions(ctx, request.NetworkIdentifier, block.Txs)
+		if err != nil {
+			return nil, service.WrapError(service.ErrInternalError, err)
+		}
+	}
+
+	for i := range transactions {
+		transaction := transactions[i]
```

Comment:
nit `	for _, transaction := range transactions {`
range returns both the index and the actual value